src/lib/data/aligned-text.js

```diff
@@ -0,0 +1,49 @@
+import Token from '@/lib/data/token'
+import TokenizeController from '@/lib/controllers/tokenize-controller.js'
+
+export default class AlignedText {
+  /**
+   *
+   * @param {SourceText} docSource
+   * @param {String} tokenizer - the name of tokenizer approach
+   */
+  constructor ({ docSource, tokenizer }) {
+    this.textType = docSource.textType
+    this.tokenizer = tokenizer
+    this.direction = docSource.direction
+    this.lang = docSource.lang
+
+    this.tokenize(docSource)
+  }
+
+  /**
+   * Defines prefix for token creations
+   */
+  get tokenPrefix () {
+    return this.textType === 'origin' ? '1' : '2'
+  }
+
+  /**
+   * Creates tokens bazed on defined method
+   * @param {SourceText} docSource
+   */
+  tokenize (docSource) {
+    const tokenizeMethod = TokenizeController.getTokenizer(this.tokenizer)
+    const tokens = tokenizeMethod(docSource.text, this.tokenPrefix, this.textType)
+    this.tokens = this.convertToTokens(tokens)
+  }
+
+  /**
+   * Formats tokens from simple objects to Token class objects
+   * @param {Array[Object]} tokens
+   */
+  convertToTokens (tokens) {
+    const tokensFormatted = []
+
+    tokens.forEach(token => {
+      const tokenFormat = new Token(token)
+      tokensFormatted.push(tokenFormat)
+    })
+    return tokensFormatted
+  }
+}

```

Comment:
@balmas , I am not sure that it is a good way to go because:

* we need to place tokens in aligned text (it is so by its sense and by the data model diagram)

* if we store tokens inside aligned texts - then it should have all the information, how these tokens were created, that's why I placed the tokenize method reference here

* also by the meaning we could have several aligned texts for one source ( I am not sure if we would want to use it), for example - we could first create alignedText by one method, than by another, and the user could choose - what would be the more useful

about your arguments:
> tokenization method could depend on SourceText
I think that in any case tokenization method would depend on user/app choice, that would depend on SourceText
We could attach variants to language or user preferences
But anyway tokenizationMethod would be used directly for creation tokens that are parts of alignedText
I think that we could have here two parameters:
* prefered tokenizationMethod (it could be defined inside Alignment object based on - preferences, availability of methods, language)
* tokenizationMethod that was used in fact - it should be placed in one place with tokens

>it will be possible to create an AlignedText from a previous Alignment, in which case the tokenization step is not going to be necessary.
I don't see problems with it at all, previously created Alignment should have information about previously used tokenizationMethod, and on upload it would be known and AlignedText would be created by upload and not by tokenize.

I believe that our misunderstood is going from differences between the data model from the diagram and my dataModel
I have the following data:

**Alignment** - all global data and methods for all states of alignment
**SourceText** - it stores text before tokenization, it would story of changes it, later it could have additional DocumentSource field for only uploaded variant from the file or request
**AlignedText** - it stores all data after starting aligning state - tokenization, tokens, alignment groups
**AlignmentGroup** - it stores data about single alignmentGroup - it could be already saved or active
**Token** - the smallest piece of data

Reply:
I'd like to think about it a bit. I think I probably haven't explained well enough the connection between a source text and an aligned text in my proposed data model, and I'm still sorting through what is really required here. I will start working on the tokenization service and through that I hope things will become clearer.